solc/CommandLineParser.cpp

```diff
@@ -1113,6 +1128,14 @@ void CommandLineParser::processArgs()
 		m_options.output.evmVersion = *versionOption;
 	}
 
+	if (m_args.count(g_strEOFVersion))
+	{
+		string versionOptionStr = m_args[g_strEOFVersion].as<string>();
+		if (versionOptionStr != "1")
+			solThrow(CommandLineValidationError, "Invalid option for --" + g_strEOFVersion + ": " + versionOptionStr);
+		m_options.output.eofVersion = 1;
```

Comment:
Why not request it as integer? Then `program_options` would validate that it's indeed an integer. This would be more consistent with how we validate it in StandardJSON.

Reply:
ok. Fine then.

Reply:
Fixed it now. Had to use a larger integer type, since ``uint8_t`` is treated as a character (i.e. I get 65 for ``--eof-version 1``).

Reply:
Nice.

I think this comment should be in the code though since it's not all that obvious why it's not `uint8_t`. I understand why it happens now that you pointed it out, but I didn't expect it :).

Reply:
Added a comment to all instances of this.